feat: non-note file enhancements

SeriousBug opened the pull request:
This PR adds a several improvements for working with non-note files.

- Links to non-note files can now link to specific lines. The syntax for this is similar to what Github uses, and it's an extension of our existing syntax. For example: `[[src/code.js#L12]]`
- Copy Note Link now works inside non-note files, and will produce a correct link to that file. If the user has a portion of the file highlighted, then a link to that line will be created using the above syntax.
- Links to non-note files are now highlighted as existing if the linked file exists
- Hovering over links to non-note files will display a "click here to open this" message, and not the incorrect "this note is missing" message.

kevinslin reviewed (changes requested):
Looks like theres a bug if note has a `.` in front. Goto note will try to open the file instead
https://www.loom.com/share/9c41b41a749f44ef8d9b5014d93c41fb

kevinslin commented on packages/plugin-core/src/commands/CopyNoteLink.ts:
```diff
@@ -29,24 +37,33 @@ export class CopyNoteLinkCommand extends BasicCommand<
+      }
     }
+    let anchor = "";
+    if (!editor.selection.isEmpty) {
```
from an UX point, when we introduce anchors, how will we diffentiate the user wanting to copy a line number vs copy an anchor?

SeriousBug replied:
I think anchors would be the default option, with a configuration to allow line numbers instead. Since line numbers would get stale too quickly, I think we should avoid encouraging their use. It'll still be an option if a user doesn't want to introduce anchors into their code base.

kevinslin replied:
sounds good. in a separate PR, can we add the following option?

```yml
commands:
  copyNoteLink:
    nonNote:
      defaultAnchorType: line|block|none
```

by default, this should be "none" until the user chooses to enable one. when we introduce the block anchor, we can switch the default to "block". I want to avoid introducing the "line" default only to switch it later.

SeriousBug commented:
Looks like it was a bug with Copy Note Link for files that are in vaults but not inside assets. Fixed and added a test for it, plus some additional tests for files/folders starting with dots.
